### ISSUE TITLE
(( used for strings

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#2727]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
aa="aa"
(( aa == "aa" )) && echo ok
(( aa == "bb" )) && echo ok

```

#### Here's what shellcheck currently says:

shellcheck accepts it without any warning

#### Here's what I wanted or expected to see:

should say (( )) is used for arithmetic evaluation, not suitable for string comparison

**Comment by aeiplatform** [comment]
```
a=10
((a == 0xA)) && echo ok
```
